### PR TITLE
Remove header and footer from signup/email confirmation pages 

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR removes the header and footer from the email confirmation and the signup pages. 

## Related Tickets & Documents
RFC: https://github.com/forem/rfcs/pull/190
Issue: https://github.com/forem/forem/issues/14649

## QA Instructions, Screenshots, Recordings

**To test the signup page:**
1. We have to prepare the db.  Open the rails console and run `User.destroy_all`, `Settings::General.mascot_user_id = nil`, `Settings::General.waiting_on_first_user = true`.
2. Thereafter, set the Feature Flag in the rails console with `FeatureFlag.enable(:creator_onboarding)`.
3. Now, go to `http://localhost:3000/enter?state=new-user`. 

You should see the sign up page without a header or footer. 
Screenshot 2021-09-27 at 15 33 43

**To test the email confirmation page:**
1. We have to prepare the db.  Open the rails console and run `User.destroy_all`, `Settings::General.mascot_user_id = nil`, `Settings::General.waiting_on_first_user = true`.
2. Set the Feature Flag in the rails console with `FeatureFlag.enable(:creator_onboarding)`.
3. Navigate to `http://localhost:3000/confirm-email?email=ridhwana%40forem.com`

You should see the sign up page without a header or footer.
 
Screenshot 2021-09-27 at 15 35 16

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [x] No, I'm not sure if I should have but these are style changes so I didn't add tests.
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, its behind a feature flag.

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by aitchiss — COMMENTED** [review]
I visited the `http://localhost:3000/confirm-email?email=ridhwana%40forem.com` url (in an incognito tab) and I see the configuration banner as well as the confirm email message - is that expected? 

screenshot of browser showing email confirmation page and the config banner

**Comment by Ridhwana** [comment]
> I visited the `http://localhost:3000/confirm-email?email=ridhwana%40forem.com` url (in an incognito tab) and I see the configuration banner as well as the confirm email message - is that expected?
> 
> screenshot of browser showing email confirmation page and the config banner

This is a good question: 

So currently, as per `verify_setup_completed` the banner will not be set on the ffg conditions:
```
return if config_path? || setup_completed? || Settings::General.waiting_on_first_user
```

However, it seems you probably have some users in your db, so you're not waiting on the first user i.e. `Settings::General.waiting_on_first_user` is false, hence the banner displays. The banner is there to deter the from creator from inviting any more users without completing setup. Hence, it will appear on all pages so they'll see it anyways.  

The use case that we're testing for is actually the **first user i.e the creator**, so I should update my instructions to say that in this scenario we will also drop the db and set waiting_for_first_user to  true, in which case the banner will not show up as expected, because `Settings::General.waiting_on_first_user`. 

TLDR; This feels like expected behaviour to me, although at some point we'll probably be rethinking the banner since we're making creator onboarding  mandatory.

**Comment by aitchiss** [comment]
OK cool - thanks @Ridhwana! Yeah I tested the creator onboarding first, so I had that one user in my DB from that point 👍

**Review by aitchiss — APPROVED** [review]

**Review by benhalpern — APPROVED** [review]
Seems good to me. A little one-offy, but seems clean enough.

**Review by citizen428 — APPROVED** [review]
